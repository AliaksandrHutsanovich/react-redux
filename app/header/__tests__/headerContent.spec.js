import React from 'react';
import { shallow } from 'enzyme';
import { Map, List } from 'immutable';
import configureStore from 'redux-mock-store';
import { Button } from 'antd';
import HeaderContent from '../headerContent';

const mockStore = configureStore();
const initialState = {
  unDoReducer: Map({
    undoOperations: List([{}, {}]),
  }),
  reDoReducer: Map({
    redoOperations: List([{}, {}]),
  }),
  countPerсent: Map({ done: 5, all: 10 }),
};

describe('Unit test of HeaderContent', () => {
  const dispatch = jest.fn();
  const store = mockStore(initialState);
  store.dispatch = dispatch;
  const Component = shallow(<HeaderContent store={store} />);

  it('full render test', () => {
    expect(Component.shallow().shallow()).toMatchSnapshot();
  });

  it('should be clickable', () => {
    Component.shallow().shallow().find(Button)
      .first()
      .simulate('click');
    expect(dispatch).toHaveBeenCalled();

    Component.shallow().shallow().find(Button)
      .at(1)
      .simulate('click');
    expect(dispatch).toHaveBeenCalled();
  });
});
