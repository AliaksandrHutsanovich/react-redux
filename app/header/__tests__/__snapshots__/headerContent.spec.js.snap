// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test of HeaderContent full render test 1`] = `
<div>
  <div
    className="headerMenu"
  >
    <h1
      className="headerLogo"
    >
      Task Accountant
    </h1>
    <ButtonGroup
      size="small"
    >
      <Button
        block={false}
        disabled={false}
        ghost={false}
        loading={false}
        onClick={[Function]}
        type="primary"
      >
        <Icon
          type="left"
        />
        Backward
      </Button>
      <Button
        block={false}
        disabled={false}
        ghost={false}
        loading={false}
        onClick={[Function]}
        type="primary"
      >
        Forward
        <Icon
          type="right"
        />
      </Button>
    </ButtonGroup>
  </div>
  <Progress
    className="progressBar"
    gapDegree={0}
    percent={50}
    showInfo={false}
    size="default"
    strokeColor="red"
    strokeLinecap="round"
    trailColor="#f3f3f3"
    type="line"
  />
</div>
`;
